src/test/java/spoon/test/imports/ImportTest.java

```diff
@@ -1861,4 +1863,28 @@ void testAutoimportConflictingSimpleNames() {
 		);
 	}
 
+	@Test
+	@GitHubIssue(issueNumber = 5210, fixed = true)
+	void staticImports_ofNestedTypes_shouldBeRecorded() {
+		Launcher launcher = new Launcher();
+		launcher.getEnvironment().setComplianceLevel(11);
+		launcher.getEnvironment().setAutoImports(true);
+		launcher.getEnvironment().setIgnoreDuplicateDeclarations(true);
+		launcher.addInputResource("src/test/resources/inner-class");
+		CtModel model = launcher.buildModel();
+
+		CtType<?> mainType = model.getElements(new TypeFilter<>(CtType.class)).stream()
+				.filter(t -> t.getSimpleName().equals("Main"))
+				.findFirst().orElseThrow();
+
+		List<CtImport> imports = mainType.getPosition().getCompilationUnit().getImports();
+		assertThat(imports, hasSize(2));
+
+		CtImport import1 = imports.get(0);
+		assertThat(import1.getReference().getSimpleName(), is("InnerClass"));
+
+		CtImport import2 = imports.get(1);
+		assertThat(import2.getReference().getSimpleName(), is("List"));
```

Comment:
I need to add two more things here:
1. Arrange act comments
2. Annotations for automatically building a model.

I will do it tomorrow. :D